packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -408,8 +433,53 @@ export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvi
                 if (nodeId && nodeId.indexOf(this.pickerOptions.elementIdPrefix) == 0) {
                     let replacementNode = this.dataProvider.onRemove(nodeAfterCursor, false);
                     this.replaceNode(nodeAfterCursor, replacementNode);
-                    this.handleKeyDownEvent(event);
+                    this.cancelDefaultKeyDownEvent(event);
+                }
+            }
+        }
+    }
+
+    private onAndroidInputEvent(event: PluginInputEvent) {
+        const wordBeforCursor = this.getInlineElementBeforeCursor(event);
+        this.newInputLength = wordBeforCursor ? wordBeforCursor.length : 0;
+
+        if (
+            this.newInputLength < this.currentInputLength ||
+            (event.rawEvent as any).inputType === DELETE_CONTENT_BACKWARDS_INPUT_TYPE
+        ) {
+            this.removeNode(event);
+        }
+
+        this.currentInputLength = this.newInputLength;
+    }
+
+    private removeNode(event: PluginDomEvent) {
+        const searcher = cacheGetContentSearcher(event, this.editor);
+        const inlineElementBefore = searcher.getInlineElementBefore();
+        const nodeBeforeCursor = inlineElementBefore
+            ? inlineElementBefore.getContainerNode()
+            : null;
+        const nodeId = nodeBeforeCursor ? this.getIdValue(nodeBeforeCursor) : null;
+        const inlineElementAfter = searcher.getInlineElementAfter();
+
+        if (
+            nodeId &&
+            nodeId.indexOf(this.pickerOptions.elementIdPrefix) == 0 &&
+            (inlineElementAfter == null || !(inlineElementAfter instanceof PartialInlineElement))
+        ) {
+            const replacementNode = this.dataProvider.onRemove(nodeBeforeCursor, true);
+            if (this.isPendingInputEventHandling) {
+                if (replacementNode) {
+                    this.replaceNode(nodeBeforeCursor, replacementNode);
+                    this.editor.runAsync(() => {

```

Comment:
By saying "only need to do this for Android", I mean the runAsync() part. For others like check null for replacementNode, call replaceNode or deleteNode, they should be fine for both.

Reply:
Oh okay, I will simplify this function then!